File: algorithms/sorting/insertion_sort.py
# sinks down the value at cursor (k)
def insertion_sort(seq):
    for k in range(1, len(seq)):
        cur = seq[k]
        j = k
        while j > 0 and seq[j-1] > cur:
            seq[j] = seq[j-1]
            j -= 1
        seq[j] = cur


arr = [2, 3, 5, 4, 2, 1, 1, 7, 6, 1, 8, 9]

insertion_sort(arr)

print(arr)

File: algorithms/sorting/insertionSort.js
'use strict';

export function insertionSort(list) {
  for (let i = 1; i < list.length; i++) {
    const up = list[i];
    let j = i - 1;
    while (j >= 0 && list[j] > up) {
        list[j + 1] = list[j]; // shift up
        j--;
    }
    list[j +  1] = up;
  }
}

// const list = [5, 9, 4, 7, 8, 6, 0, 1, 3, 2];
// insertionSort(list);
// console.log(list);